common/scripts/setup_env.sh

```diff
@@ -163,21 +168,67 @@ fi
 
 KUBECONFIG=${KUBECONFIG:="$HOME/.kube/config"}
 parse_KUBECONFIG "${KUBECONFIG}"
-if [[ "${BUILD_WITH_CONTAINER:-1}" -eq "1" ]]; then
-  export KUBECONFIG="${container_kubeconfig%?}"
+if [[ "${FOR_BUILD_CONTAINER:-0}" -eq "1" ]]; then
+  KUBECONFIG="${container_kubeconfig%?}"
 fi
 
-# Avoid recursive calls to make from attempting to start an additional container
-export BUILD_WITH_CONTAINER=0
+# LOCAL_OUT should point to architecture where we are currently running versus the desired.
+# This is used when we need to run a build artifact during tests or later as part of another
+# target.
+if [[ "${FOR_BUILD_CONTAINER:-0}" -eq "1" ]]; then
+  LOCAL_OUT="${TARGET_OUT_LINUX}"
+else
+  LOCAL_OUT="${TARGET_OUT}"
+fi
+
+if [[ "${FOR_BUILD_CONTAINER:-0}" -eq "1" ]]; then
+  # Override variables with container specific
+  TARGET_OUT=${CONTAINER_TARGET_OUT}
+  TARGET_OUT_LINUX=${CONTAINER_TARGET_OUT_LINUX}
+  REPO_ROOT=/work
+fi
+
+go_os_arch=${LOCAL_OUT##*/}
+# Golang OS/Arch format
+LOCAL_GO_OS=${go_os_arch%_*}
+LOCAL_GO_ARCH=${go_os_arch##*_}
+
+BUILD_WITH_CONTAINER=0
```

Comment:
Not sure if we want this overridable?

Reply:
My expectation is you are either:

1. out of build container and setting up env, and planning to stay out of the build container
2. Out of the build container and setting up, but preparing to go into the container. FOR_BUILD_CONTAINER=1 is set. Once you are in the container, you do not want nested container, so BUILD_WITH_CONTAINER=0 is set.
It was the same prior to this PR btw

Reply:
A reason it may not need to be is that `BUILD_WITH_CONTAINER=1 make clean` does require the build-tools image while being 0 did not.

Reply:
Not sure I understand. If you run `BUILD_WITH_CONTAINER=1 make clean` then:

* `./common/scripts/run.sh make clean` is run
* This sources setup_env with FOR_BUILD_CONTAINER=1
* Env is setup with all container specific things, and BUILD_WITH_CONTAINER=0
* Any future calls to make will have BUILD_WITH_CONTAINER=0 and not do nested containers

That is how I would expect it to work. What do you expect?

Reply:
That is what I expected. I was hoping that the BUILD_WITH_CONTAINER=1 wouldn't have been overriden and not happened in a container at all. It appears that it wasn't.